silence/silence.go

```diff
@@ -276,6 +325,11 @@ func (s *Silences) setSilence(sil *pb.Silence) error {
 	if err != nil {
 		return err
 	}
+
+	_, err = s.mc.Add(sil)
+	if err != nil {
+		return err
+	}
```

Comment:
I feel like there are lots of edge cases which can cause the cache to miss, but "pre-heating" the cache on `setSilence` and `loadSnapshot` still seems a good idea, wdyt @fabxc ?